agent/api/container/container.go

```diff
@@ -1231,14 +1201,34 @@ func (c *Container) GetTaskARN() string {
 	return c.TaskARNUnsafe
 }
 
-func (c *Container) SetExecCommandAgentMetadata(metadata ExecCommandAgentMetadata) {
-	c.lock.Lock()
-	defer c.lock.Unlock()
-	c.ExecCommandAgentMetadata = metadata
-}
-
-func (c *Container) GetExecCommandAgentMetadata() ExecCommandAgentMetadata {
+// GetManagedAgentByName retrieves the managed agent with the name specified and a boolean indicating whether an agent
+// was found or not.
+// note: a zero value for ManagedAgent if the name is not known to this container.
+func (c *Container) GetManagedAgentByName(agentName string) (ManagedAgent, bool) {
 	c.lock.RLock()
 	defer c.lock.RUnlock()
-	return c.ExecCommandAgentMetadata
+	for _, ma := range c.ManagedAgentsUnsafe {
+		if ma.Name == agentName {
+			return ma, true
+		}
+	}
+	return ManagedAgent{}, false
+}
+
+// UpdateManagedAgentByName updates the state of the managed agent with the name specified. If the agent is not found,
+// no action is taken.
```

Comment:
should we atleast log if agent is not found? 
when do we anticipate such an issue?

Reply:
Sure, we can log, will update.
For example, if `UpdateManagedAgentByName` is called on an internal container, it surely won't have any managed agent.

Reply:
Actually I ended up returning a boolean depending on whether the agent is found, that way is up to the caller to log or do the necessary